openc3-cosmos-cmd-tlm-api/app/controllers/queues_controller.rb

```diff
@@ -0,0 +1,285 @@
+      ret = Array.new
+      queues.each do |_, trigger|
+        ret << trigger
+      end
+      render json: ret
+    rescue StandardError => e
+      log_error(e)
+      render json: { status: 'error', message: e.message, type: e.class.to_s, backtrace: e.backtrace }, status: 500
+    end
+  end
+
+  # Returns a queue in json.
+  #
+  # name [String] the queue name, `QUEUE0`
+  # scope [String] the scope of the queue, `TEST`
+  # @return [String] the queue converted into json format.
+  def show
+    return unless authorization('cmd_info')
+    begin
+      model = @model_class.get(name: params[:name], scope: params[:scope])
+      if model.nil?
+        render json: { status: 'error', message: NOT_FOUND }, status: 404
+        return
+      end
+      render json: model.as_json(:allow_nan => true)
+    rescue OpenC3::QueueError => e
+      log_error(e)
+      render json: { status: 'error', message: e.message, type: e.class.to_s }, status: 400
+    rescue StandardError => e
+      log_error(e)
+      render json: { status: 'error', message: e.message, type: e.class.to_s, backtrace: e.backtrace }, status: 500
+    end
+  end
+
+  # Create a new queue and return the object/hash of the trigger in json.
+  #
+  # name [String] the queue name, `QUEUE0`
+  # scope [String] the scope of the trigger, `TEST`
+  def create
+    return unless authorization('cmd')
+    begin
+      model = @model_class.get(name: params[:name], scope: params[:scope])
+      if model.nil?
+        state = 'HOLD'
+        state = params[:state] if params[:state]
+        model = @model_class.new(name: params[:name], state: state, scope: params[:scope])
+        model.create()
+        render json: model.as_json(:allow_nan => true), status: 201
+      else
+        render json: { status: 'error', message: "#{params[:name]} already exists", type: "OpenC3::QueueError" }, status: 400
+      end
+    rescue OpenC3::QueueError => e
+      log_error(e)
+      render json: { status: 'error', message: e.message, type: e.class.to_s }, status: 400
+    rescue StandardError => e
+      log_error(e)
+      render json: { status: 'error', message: e.message, type: e.class.to_s, backtrace: e.backtrace }, status: 500
+    end
+  end
+
+  def list
+    return unless authorization('cmd_info')
+    begin
+      model = @model_class.get_model(name: params[:name], scope: params[:scope])
+      if model.nil?
+        render json: { status: 'error', message: NOT_FOUND }, status: 404
+        return
+      end
+      list = model.list()
+      render json: list
+    rescue StandardError => e
+      log_error(e)
+      render json: { status: 'error', message: e.message, type: e.class.to_s, backtrace: e.backtrace }, status: 500
+    end
+  end
+
+  def hold
+    change_state(params, 'HOLD')
+  end
+
+  def release
+    change_state(params, 'RELEASE')
+  end
+
+  def disable
+    change_state(params, 'DISABLE')
+  end
+
+  def insert_command
+    return unless authorization('cmd')
```

Comment:
Need to check cmd permissions for specific target/packet.  Applies throughout.

Reply:
Throughout = everywhere where there is a specific command.   Creating a queue would not apply